ckanext/qgov/common/anti_csrf.py

```diff
@@ -94,6 +83,35 @@ def get_cookie_token():
 
     return token
 
+def validate_token(token):
+    if not HMAC_PATTERN.match(token):
+        return false
+
+    import time, hmac, hashlib, urllib
+
+    now = int(time.time())
+
+    parts = token.split('!', 1)
+    provided_hmac = unicode(parts[0])
+    message = parts[1]
+
+    secret_key = _get_secret_key()
+    expected_hmac = unicode(hmac.HMAC(secret_key, message, hashlib.sha512).hexdigest())
+    if not hmac.compare_digest(expected_hmac, provided_hmac):
+        return False
+
+    message_parts = message.split('/', 2)
+    timestamp = int(message_parts[0])
+    username = message_parts[2]
```

Comment:
you should drop / if it has it to match generator. better yet, have a method that generates the message, and a method that spits out an array of what was in the message thats standard.